### PR TITLE
Adding SAMP/T Block 1, Block 1NT and Block 2 (and other changes)

### PR DESCRIPTION
**Pull request by zahnatom**
I was bored
So idk I wrote the code for this, the models and textures were already there

no guarantees for accurate performance though

### CODE REVIEW

File: Mods/tech/HighDigitSAMs/Database/Weapon/ASTER-30-1.lua [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    Reflection 		= 0.1,
+    KillDistance 	= 20.0,
+	tail_scale 	 	= 1.4,		
+	ccm_k0 			= 0.2,	
+	
+	active_radar_lock_dist	= 150000.0,
+	go_active_by_default	= 0,	
+
+	PN_coeffs = {3, 				-- Number of Entries
+				50000.0 ,1.0,		-- Less 5 km to target Pn = 1
```
Comment (YoloWingPixie) [inline Mods/tech/HighDigitSAMs/Database/Weapon/ASTER-30-1.lua:55]: You have 50km set here, not 5km. I do not think you want to be full lead guidance at 50km

File: Mods/tech/HighDigitSAMs/Database/Weapon/ASTER-30-1.lua [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+declare_weapon(SAMPT_ASTER_30_Blk_1)
+
+GT_t.LN_t.Aster30_1 = {}; 
+GT_t.LN_t.Aster30_1.type = 4;
+GT_t.LN_t.Aster30_1.distanceMin = 1000;
+GT_t.LN_t.Aster30_1.distanceMax = 120000;
+GT_t.LN_t.Aster30_1.reactionTime = 0.1;
+GT_t.LN_t.Aster30_1.launch_delay = 0.1;
+GT_t.LN_t.Aster30_1.reflection_limit = 0.0001;
+GT_t.LN_t.Aster30_1.ECM_K = 0
```
Comment (YoloWingPixie) [inline Mods/tech/HighDigitSAMs/Database/Weapon/ASTER-30-1.lua:83]: Again, I believe this should be 0.1 or 0.2, not a flat 0.

File: Mods/tech/HighDigitSAMs/Database/Weapon/ASTER-30-1NT.lua [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    Diam 			= 180.0,
+    Cx_pil 			= 1,
+    D_max 			= 100000.0,
+    D_min 			= 2000.0,
+    Head_Form 		= 1,
+    Life_Time 		= 360.0,
+    Nr_max 			= 60,
+    v_min 			= 50.0,
+    v_mid 			= 1800.0,
+    Mach_max 		= 5.0,
```
Comment (YoloWingPixie) [inline Mods/tech/HighDigitSAMs/Database/Weapon/ASTER-30-1NT.lua:27]: Same question as the Blk1, where does this value come from?

File: Mods/tech/HighDigitSAMs/Database/Weapon/ASTER-30-1NT.lua [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+declare_weapon(SAMPT_ASTER_30_Blk_1NT)
+
+GT_t.LN_t.Aster30_1NT = {}; 
+GT_t.LN_t.Aster30_1NT.type = 4;
+GT_t.LN_t.Aster30_1NT.distanceMin = 1000;
+GT_t.LN_t.Aster30_1NT.distanceMax = 150000;
+GT_t.LN_t.Aster30_1NT.reactionTime = 0.1;
+GT_t.LN_t.Aster30_1NT.launch_delay = 0.1;
+GT_t.LN_t.Aster30_1NT.reflection_limit = 0.0001;
+GT_t.LN_t.Aster30_1NT.ECM_K = 0
```
Comment (YoloWingPixie) [inline Mods/tech/HighDigitSAMs/Database/Weapon/ASTER-30-1NT.lua:83]: Same comment as before, 0.1/0.2 not 0.

File: Mods/tech/HighDigitSAMs/Database/Weapon/ASTER-30-2.lua [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+    Reflection 		= 0.1,
+    KillDistance 	= 20.0,
+	tail_scale 	 	= 1.4,		
+	ccm_k0 			= 0.2,	
+	
+	active_radar_lock_dist	= 300000.0,
+	go_active_by_default	= 0,	
+
+	PN_coeffs = {3, 				-- Number of Entries
+				100000.0 ,1.0,		-- Less 5 km to target Pn = 1
```
Comment (YoloWingPixie) [inline Mods/tech/HighDigitSAMs/Database/Weapon/ASTER-30-2.lua:55]: Here we have full lead guidance being implemented at 100km, I do not think this is what you intend. That is a lot of wasted energy if the target manuevers.

File: Mods/tech/HighDigitSAMs/Database/Weapon/ASTER-30-1.lua [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+    category        = CAT_MISSILES,
+    name            = nameaster301,
+    user_name       = _(nameaster301),
+    model           = 'aster_30_blk_1',
+    mass            = 450.0,
+    wsTypeOfWeapon  =  {wsType_Weapon,wsType_Missile,wsType_SA_Missile,WSTYPE_PLACEHOLDER},
+
+    Escort 			= 0,
+    Head_Type 		= 2,
+	sigma 			= {10, 10, 10},
```
Comment (YoloWingPixie) [inline Mods/tech/HighDigitSAMs/Database/Weapon/ASTER-30-1.lua:14]: Seems kind of low, I think `{20, 20, 20}` would be a little more representative

File: Mods/tech/HighDigitSAMs/Database/Weapon/ASTER-30-1NT.lua
Comment (YoloWingPixie) [inline Mods/tech/HighDigitSAMs/Database/Weapon/ASTER-30-1NT.lua:1]: Consistent whitespace issues here too

File: Mods/tech/HighDigitSAMs/Database/Weapon/ASTER-30-2.lua
Comment (YoloWingPixie) [inline Mods/tech/HighDigitSAMs/Database/Weapon/ASTER-30-2.lua:1]: Consistent whitespace issues

**Review by YoloWingPixie — CHANGES_REQUESTED** [review]
I've added my comments to this, looks very good. Some small things I'd like to see worked out personally.

**Comment by YoloWingPixie** [comment]
Might have been a display issue, it looks like most of the whitespace is justified now. 

I or @Auranis will test again soon

**Comment by YoloWingPixie** [comment]
Yep, we've encountered the same thing with the SA-20 missiles. It was fixable for us by switching over to `PN_Gain` and then cranking it up until the missile barely stays below the altitude limit before it'll escape into space with a worst case target.

**Comment by YoloWingPixie** [comment]
Please undo that change, that's already in another pull request

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by YoloWingPixie** [comment]
Please undo the changes to the 48N6 and 48N6E2, there's already changes to those missiles in #53

**Comment by Auranis** [comment]
Thanks for your implementation @zahnatom we will include this in the upcoming release.

**Pull request merged**